test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,352 @@
+		map[string]string{customDnsLabelKey: customDnsLabelValue}, false, false, corev1.ServiceTypeClusterIP, ptr.To[corev1.IPFamily](corev1.IPv4Protocol), map[string]string{})
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	setDnsServerAddressInAntrea(t, data, customDnsService.Spec.ClusterIP)
+	defer setDnsServerAddressInAntrea(t, data, "")
+
+	podCount := 2
+	agnHostPodIps := make([]*PodIPs, podCount)
+	for i := 0; i < podCount; i++ {
+		agnHostPodIps[i] = createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	}
+
+	ipv4, ipv6 := extractIPs(agnHostPodIps[0])
+
+	dnsConfigData := createDnsConfig(t, ipv4, ipv6, testFullyQualifiedDomainName)
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      customDnsConfigName,
+			Namespace: data.testNamespace,
+		},
+		Data: dnsConfigData,
+	}
+	customDnsConfigMapObject, err := data.CreateConfigMap(configMap)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createDnsPod(t, data)
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	require.NoError(t, err, "error getting k8s utils %+v", err)
+
+	createFqdnPolicyInNamespace(t, data, testFullyQualifiedDomainName)
+
+	t.Logf("Creating Toolbox pod...")
+	createToolBoxPod(t, data, customDnsService.Spec.ClusterIP)
+
+	curlTarget := func(podName, containerName, fqdn string) (string, error) {
+		cmd := []string{"curl", fqdn}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, podName, containerName, cmd)
+		if err != nil {
+			return "", fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return stdout, nil
+	}
+
+	t.Logf("Trying to curl FQDN ...")
+	assert.EventuallyWithT(t, func(collect *assert.CollectT) {
+		stdout, err := curlTarget(toolboxPodName, toolboxContainerName, testFullyQualifiedDomainName)
+		assert.NoError(t, err)
+		t.Logf("response of curl to FQDN - %+v", stdout)
+	}, 2*time.Second, 100*time.Millisecond, "trying to curl the fqdn : ", testFullyQualifiedDomainName)
+
+	t.Logf("Trying to DIG the FQDN to simulate caching the current IP inside Toolbox pod...")
+	digResponse, err := k8sUtils.runDNSQuery(toolboxPodName, toolboxContainerName, data.testNamespace, testFullyQualifiedDomainName, false, customDnsService.Spec.ClusterIP)
+	require.NoError(t, err, "failed to get IP of FQDN using DIG from toolbox pod : %v", err)
+
+	// Use an assertion to check if the received IPs are same.
+	fqdnIp := digResponse.String()
+	if digResponse.To4() == nil {
+		require.Equalf(t, ipv6, fqdnIp, "The IP set against the FQDN in the DNS server should be the same, but got %+v instead of %+v", fqdnIp, ipv6)
+	} else {
+		require.Equalf(t, ipv4, fqdnIp, "The IP set against the FQDN in the DNS server should be the same, but got %+v instead of %+v", fqdnIp, ipv4)
+	}
```

Comment:
this is not quite right. For the dual-stack case, we should have both IPs. This may imply running `dig` twice, to send both an A query and an AAAA query. Since supporting dual-stack is a bit of a pain point right now, maybe we should start with a test that only supports IPv4. We can add a call to `skipIfNotIPv4Cluster` and a call to `skipIfIPv6Cluster` at the beginning of the test. After that you can assume that all IP addresses are IPv4 addresses.

Reply:
to add to this: it will be easier to add correct support for IPv6 / dual-stack _after_ the `minTTL` functionality is implemented and we expect the test to pass. At that point, we can handle adding that support in a separate PR.